### PR TITLE
Enable permission and authentication to service calls 

### PR DESCRIPTION
**Pull request by Kamidu**

### CODE REVIEW

**Comment by boxidilshan** [comment]
Merging pull request

File: modules/apps/jaggery/mdm/src/mdm/modules/backendServiceInvoker.js [hunk 1]
```diff
@@ -135,14 +135,19 @@ var backendServiceInvoker = function () {
     function initiateWSRequest(action, endpoint, payload, successCallback, errorCallback, soapVersion) {
         var ws = require('ws');
         var wsRequest = new ws.WSRequest();
-        var options = [];
+        var options = new Array();
         if (IS_OAUTH_ENABLED) {
             var accessToken = session.get(constants.ACCESS_TOKEN_PAIR_IDENTIFIER).accessToken;
-            if (!(!accessToken.trim())) {
-                var headers = [{name: constants.AUTHORIZATION_HEADER, value: constants.BEARER_PREFIX + accessToken}];
+            if (!(!accessToken)) {
```
Comment (harshanL) [inline modules/apps/jaggery/mdm/src/mdm/modules/backendServiceInvoker.js]: what is the point of having !! here in (!(!accessToken)) ?

Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/modules/backendServiceInvoker.js]: Recommended fixes  can be found in PR[1]

[1]. https://github.com/wso2/product-mdm/pull/158

File: modules/apps/jaggery/mdm/src/mdm/modules/policy.js [hunk 2]
```diff
@@ -73,14 +68,15 @@ policyModule = function () {
...
                 // generate response
-                response["status"] = "success";
-                response["content"] = policyListToView;
+                response.status = "success";
+                response.content = policyListToView;
                 return response;
-            } else {
-                // generate response
-                response["status"] = "error";
+            },function(){
```
Comment (prabathabey) [inline modules/apps/jaggery/mdm/src/mdm/modules/policy.js]: Fix formatting issues.

File: modules/distribution/src/repository/conf/security/authenticators.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (prabathabey) [inline modules/distribution/src/repository/conf/security/authenticators.xml]: License header format used is obsolete. Please fix.

File: modules/apps/jaggery/mdm/src/mdm/modules/policy.js (hunk 2 above)
Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/modules/policy.js]: These formatting issues has been fixed in a recent PR

File: modules/distribution/src/repository/conf/security/authenticators.xml (hunk 3 above)
Comment (Kamidu) [inline modules/distribution/src/repository/conf/security/authenticators.xml]: Recommended fixes can be found in PR[1]

[1]. https://github.com/wso2/product-mdm/pull/166

**Pull request merged**